public/app.js

```diff
@@ -52,78 +57,47 @@ const generateComputerChoice = () => {
     }
-    
+
     return player.choice
 }
+const displayResult = (playerChoice, computerChoice, result) => {
+    resultDisplay.innerHTML = `Your element ${playerChoice} ${result} against computer element ${computerChoice}`;
+}
 
-//Compare choices/selectwinner
-const selectWinner = () =>{
-    let result = null;
-    generateComputerChoice();
-    getPlayerChoice();
 
-    if (player.choice !== null) {
-        switch(player.choice+computer.choice){
-            //Win choice
-            case "firewood":
-            case "firemetal":
-            case "earthfire":
-            case "earthwater":
-            case "woodearth":
-            case "woodwater":
-            case "metalwood":
-            case "metalearth":
-            case "waterfire":
-            case "watermetal":
-                result = "win";
-                score++;
-                round++;
-                resultDisplay.innerHTML = `Your element ${player.choice} ${result} against computer element: ${computer.choice}`;
-                break;
-            //Lose choice
-            case "firewater":
-            case "fireearth":
-            case "earthwood":
-            case "earthmetal":
-            case "woodfire":
-            case "woodmetal":
-            case "metalfire":
-            case "metalwater":
-            case "waterwood":
-            case "waterearth":
-                result = "lose";
-                round++;
-                resultDisplay.innerHTML = `Your element ${player.choice} ${result} against computer element: ${computer.choice}`;
-                break;
-            //Draw choice
-            case "firefire":
-            case "earthearth":
-            case "woodwood":
-            case "metalmetal":
-            case "waterwater":
-                result = "draw";
-                round++;
-                resultDisplay.innerHTML = `Your element ${player.choice} ${result} against computer element: ${computer.choice}`;
-                break;
-        }
-    }else{
-        resultDisplay.innerHTML = `Select your element to play!`
-    }
-    scoreDisplay.innerHTML = score + "/" + round;
-
-//Update score & highscore - localstorage
+const selectWinner = () => {
     if (round === 20) {
         if (score > highScoreStorage.getItem('highScore')) {
             highScoreStorage.setItem('highScore', score)
         }
         resetGame();
+    }else{
+        getPlayerChoice();
+        generateComputerChoice();
+
+        pairs
+    .then((value =>{
+        let result = value.default[0].pairs[player.choice][computer.choice]
+        if (result === true) {
+            displayResult(player.choice, computer.choice, "won")
+            score++;
+            round++;
+        } else if (result === false) {
+            displayResult(player.choice, computer.choice, "lose")
+            round++;
+        } else {
+            displayResult(player.choice, computer.choice, "draw")
+            round++;
+        }
```

Comment:
Jak pozbędziesz się promise'a z pairs, to będzie można pozbyć się `then`.

- nie nadpisujesz `result` więc `const` raczej a nie `let`.
- z `if (result === true)` to bym polemizował 😄 
- `round++` występuje w każdym ifie, więc można spokojnie dać to na koniec.

```ts
  const result = pairs[player.choice][computer.choice];
  if (result === null) displayResult(player.choice, computer.choice, "draw");
  else if (result) displayResult(player.choice, computer.choice, "won");
  else displayResult(player.choice, computer.choice, "lose");
  round++;
  scoreDisplay.innerHTML = score + "/" + round;
```